docs/decisions/0004-exam-ida.rst

```diff
@@ -0,0 +1,59 @@
+
+We believe we can create a well-defined interface that results in an exam solution that is best managed as its own service. The following proposed component diagram defines the relationship between platform components and the bounded context of timed exams.
+
+`Exam Components`_
+
+Additionally, we intend to eliminate vendor-specific integration code by implementing the IMS's `Proctoring Services Standard`_ built on top of LTI 1.3. This new IDA gives us a clean starting point to build that interface.
+
+.. _Proctoring Services Standard: https://www.imsglobal.org/proctoring-services
+
+.. _Exam Components: https://lucid.app/lucidchart/a4b40637-93f1-47f1-bb17-d68e4ff6f9d9/edit?invitationId=inv_92a6d5c4-d14c-472a-b2bc-64b64d9d45ef
+
+Decision
+--------
+A new repository and IDA will be created for the exam service. As service will handle all configuration and interfacing with third party proctoring tools. It will also be the source of truth for any state information about a learner's progress through exam completion and review flows.
+
+* The IDA will function in parallel with the current edx-proctoring library. edx-platform may use either service for a course.
+
+* The LMS will not depend on this service. The exams service will push any relevant state to the platform.
+
+* Changes in exam state that would impact other systems such as grades, completion, credit, or certificates should be pushed to those systems via REST endpoints or using events.
+
+* Data about course content or exam configuration will be pushed to the IDA's REST API as part of the studio publish action. We should replicate data instead of reading directly from the CMS or LMS in an ad-hoc manner.
+
+* The IDA will implement a REST API to expose exam and attempt state information to https://github.com/edx/frontend-lib-special-exams.
+
+* The IDA will not include custom APIs or data calls for specific to a single proctoring tool.
+
+Consequences
+------------
+
+* The display of learner attempt state in course outlines requires the LMS to directly depend on the proctoring service. This feature will be simplified to only contain information about the type of exam. Attempt state will be accessible within the section via the frontend-lib-special-exams library.
+
+* The LMS will need a new way to prevent rendering exam content while not activly in an exam that does not rely on the installation of this library.
+
+* Existing proctoring vendors will remain functional using this library but will not by-default be implemented in the new IDA.
+
+* We will need to maintain both proctoring implementations until we can migrate use away from this library and the vendors it specifically supports.
+
+
```

Comment:
I don't think there ever was an alternative here. The #1 goal was to address the independent rate of change between this code and edx-platform, and ultimately not being tied to platform deployments. Another plugin doesn't really solve this. I probably need to underscore that better since a lot of this doc does just go off and talk about the IDA interface.